### PR TITLE
sharepoint webURL selector scopes

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds webUrl scopes to the sharepoint selector.
Also introduces the idea of a scope categories
that can broadly match across all leaf types
within a service.  This category union should be
reserved for root categories, and properties
that can be used interchangably with the root.

This is part 2 of exposing webURLs as an alternative
to siteIDs for sharepoint backup and restore.  The
next change providing a webURL => siteID lookup
within the graph package.

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #1616

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Comment by ashmrtn** [comment]
this appears to need rebasing as the diff has changes from the suffix filter patches

**Comment by ryanfkeepers** [comment]
@ashmrtn Thanks for the patience.  Rebase is complete, the diff should be clean.

**Review by ashmrtn — APPROVED** [review]

File: src/pkg/selectors/selectors.go [hunk 1]
```diff
@@ -515,6 +549,14 @@ func wrapFilter(ff filterFunc) func([]string) filters.Filter {
...
 		}
 
+		if cfg.usePathSuffixFilter {
+			return filters.PathSuffix(s)
+		}
+
+		if cfg.usePathContainsFilter {
+			return filters.PathContains(s)
+		}
+
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/selectors/selectors.go:559]: mmm, that's a good call-out.  I'll withdraw the options handling here in favor of passing in a filter, and if things get clunky from pushing that resolution up a level then we can address that separately.

[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/selectors/selectors.go]: @ashmrtn updated.  See how you like the new approach.

**Pull request merged**